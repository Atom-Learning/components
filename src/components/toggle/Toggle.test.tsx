import { render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import React from 'react'

import { Toggle } from '.'

describe(`Toggle component`, () => {
  it('renders a toggle switch', async () => {
    const { container } = render(<Toggle />)

    expect(container).toMatchSnapshot()
  })

  it('renders a toggle - has no programmatically detectable a11y issues', async () => {
    const { container } = render(<Toggle aria-label="toggle" />)
    expect(await axe(container)).toHaveNoViolations()
  })

  it('renders a toggle switch', async () => {
    const { container } = render(<Toggle disabled />)

    expect(container).toMatchSnapshot()
  })

  it('renders a disabled toggle - has no programmatically detectable a11y issues', async () => {
    const { container } = render(<Toggle disabled aria-label="toggle" />)
    expect(await axe(container)).toHaveNoViolations()
  })
})
