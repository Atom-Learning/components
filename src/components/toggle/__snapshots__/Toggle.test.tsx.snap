// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle component renders a toggle switch 1`] = `
.sx6gk47 {
  -webkit-appearance: none;
  appearance: none;
  background-color: var(--sx-colors-tonal300);
  border: none;
  padding: 2px;
  box-shadow: inset 0 0 0 1px var(--sx-shadows-tonal300);
  overflow: hidden;
  border-radius: var(--sx-radii-round);
  cursor: pointer;
  display: inline-flex;
  width: 50px;
}

.sx6gk47:focus {
  outline: none;
}

.sx6gk47[data-state=on] {
  background-color: var(--sx-colors-success);
  justify-content: flex-end;
}

.sx6gk47[data-state=off] {
  justify-content: flex-start;
}

.sxz551g {
  height: 20px;
  width: 20px;
  background: white;
  display: inline-block;
  border-radius: var(--sx-radii-round);
}

<div>
  <button
    aria-pressed="false"
    class="sx6gk47"
    data-state="off"
    type="button"
  >
    <span
      class="sxz551g"
    />
  </button>
</div>
`;

exports[`Toggle component renders a toggle switch 2`] = `
.sx6gk47 {
  -webkit-appearance: none;
  appearance: none;
  background-color: var(--sx-colors-tonal300);
  border: none;
  padding: 2px;
  box-shadow: inset 0 0 0 1px var(--sx-shadows-tonal300);
  overflow: hidden;
  border-radius: var(--sx-radii-round);
  cursor: pointer;
  display: inline-flex;
  width: 50px;
}

.sx6gk47:focus {
  outline: none;
}

.sx6gk47[data-state=on] {
  background-color: var(--sx-colors-success);
  justify-content: flex-end;
}

.sx6gk47[data-state=off] {
  justify-content: flex-start;
}

.sxz551g {
  height: 20px;
  width: 20px;
  background: white;
  display: inline-block;
  border-radius: var(--sx-radii-round);
}

<div>
  <button
    aria-pressed="false"
    class="sx6gk47"
    data-disabled=""
    data-state="off"
    disabled=""
    type="button"
  >
    <span
      class="sxz551g"
    />
  </button>
</div>
`;
